LBH.AdultSocialCare.Functions.Payruns/LambdaEntryPoint.cs

```diff
@@ -12,41 +12,33 @@
 
 namespace LBH.AdultSocialCare.Functions.Payruns
 {
+#pragma warning disable CA1052
     public class LambdaEntryPoint
     {
-        private readonly ILogger<LambdaEntryPoint> _logger;
-        private readonly PayrunGenerator _payrunGenerator;
-
-        // AWS entry point
-        // ReSharper disable once UnusedMember.Global
-        public LambdaEntryPoint() : this(InitServices())
-        {
-        }
-
-        // Local runner / unit tests entry point
-        public LambdaEntryPoint(IServiceProvider services)
+        [LambdaSerializer(typeof(JsonSerializer))]
+        public static async Task HandleEvent(SQSEvent sqsEvent)
         {
-            _logger = services.GetService<ILogger<LambdaEntryPoint>>();
-
+            // Lambda constructor is called just one time so create services on each request
+            // to ensure that scoped services lifetime (especially DBContext) is short
+            var services = InitServices();
             var serviceScopeFactory = (IServiceScopeFactory) services.GetService(typeof(IServiceScopeFactory));
-            var serviceScope = serviceScopeFactory.CreateScope();
 
-            _payrunGenerator = serviceScope.ServiceProvider.GetRequiredService<PayrunGenerator>();
-        }
+            using (var serviceScope = serviceScopeFactory.CreateScope())
+            {
+                var logger = services.GetRequiredService<ILogger<LambdaEntryPoint>>();
+                var payrunGenerator = serviceScope.ServiceProvider.GetRequiredService<PayrunGenerator>();
 
-        [LambdaSerializer(typeof(JsonSerializer))]
-        public async Task HandleEvent(SQSEvent sqsEvent)
-        {
-            _logger.LogInformation("Received SQS event {SqsEvent}", JsonConvert.SerializeObject(sqsEvent, Formatting.Indented));
+                logger.LogInformation("Received SQS event {SqsEvent}", JsonConvert.SerializeObject(sqsEvent, Formatting.Indented));
 
-            try
-            {
-                await _payrunGenerator.GenerateAsync();
-            }
-            catch (Exception ex)
-            {
-                _logger.LogError("Unhandled exception {Exception}", ex);
-                throw;
+                try
+                {
+                    await payrunGenerator.GenerateAsync();
+                }
+                catch (Exception ex)
+                {
```

Comment:
exception handling should be handle into the payrun generator too. When something going wrong for payrun, it should be logged and changed status to "archived"

Reply:
this is not implemented yet, in my TBD, will implement it in the end, since it's minor thing comparing to the rest